functions-continued/variables-are-references.md

```diff
@@ -0,0 +1,648 @@
+1. Reassignment changes what the variable points to
+
+```python
+berries = ["hearts", "stars", "horseshoes"]
+melons = berries
+loquats = melons
+print("id of berries:", id(berries))
+print("id of melons:", id(melons))
+print("id of loquats:", id(loquats))
+
+print("Reassigning melons to a new object:")
+
+melons = ["sneezy", "sleepy", "happy"]
+print("id of berries:", id(berries))
+print("id of melons:", id(melons))
+print("id of loquats:", id(loquats))
+
+print("Reassigning melons to berries (again):")
+
+melons = berries
+print("id of berries:", id(berries))
+print("id of melons:", id(melons))
+print("id of loquats:", id(loquats))
+```
+
+```
+id of berries: 4510649216
+id of melons: 4510649216
+id of loquats: 4510649216
+Reassigning melons to a new object:
+id of berries: 4510649216
+id of melons: 4510624000
+id of loquats: 4510649216
+Reassigning melons to berries (again):
+id of berries: 4510649216
+id of melons: 4510649216
+id of loquats: 4510649216
+```
+
+### !callout-success
+
+## Pro-Tip: Experiment!
+
+There are an infinite number of ways to assign and reassign and check this logic. Experiment in your own Python files!
+
+### !end-callout
+
+## Modifying Mutable Objects
+
+Let's review our major concepts so far:
+
+1. Variables are references to objects in memory
+1. Immutable data types don't change. Different immutable objects always occupy different places in memory and have different object IDs.
+1. Mutable data types can change. Even if an object is modified, it will still occupy the same place in memory and have the same object ID.
+1. Multiple variables can point to the same object
+
+When we combine and synthesize these concepts, we can observe that **multiple references to the same mutable object will all see the same changes**.
+
+```python
+berries = ["hearts", "stars", "horseshoes"]
+melons = berries
+
+print("Before modifying berries")
+print("id of berries:", id(berries))
+print("id of melons:", id(melons))
+print("value of melons:", melons)
+
+berries.append("clovers")
+
+print("After modifying berries")
+print("id of berries:", id(berries))
+print("id of melons:", id(melons))
+print("value of berries:", berries)
+print("value of melons:", melons)
+```
+
+When we run the above code, we may see output like this:
+
+```
+Before modifying berries
+id of berries: 4369938880
+id of melons: 4369938880
+value of melons: ['hearts', 'stars', 'horseshoes']
+After modifying berries
+id of berries: 4369938880
+id of melons: 4369938880
+value of berries: ['hearts', 'stars', 'horseshoes', 'clovers']
+value of melons: ['hearts', 'stars', 'horseshoes', 'clovers']
+```
+
+Re-read through the code slowly, and take notes about how the modification to `berries` affects the value that `melons` points to.
+
+`berries` and `melons` both point to the same object. We modify the object with object ID `4369938880` with `berries.append("clovers")`. We see that the value of `melons` (which points to the object `4369938880`) is also modified.
```

Comment:
Add also to the parenthetical for reinforcement, and then rephrase the following 'also' so it doesn't sound repetitive.

```suggestion
`berries` and `melons` both point to the same object. We modify the object with object ID `4369938880` with `berries.append("clovers")`. We see that the value of `melons` (which also points to the object `4369938880`) is modified as well.
```